src/main/java/bhs/devilbotz/commands/drivetrain/DriveVision.java

```diff
@@ -0,0 +1,130 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import org.photonvision.PhotonCamera;
+
+/** This command is a PID controller that drives the robot straight to a set distance. */
+public class DriveVision extends CommandBase {
+  protected DriveTrain drivetrain;
+  private PIDController distancePid;
+  private PIDController straightPid;
+  private double distance;
+  private double startAngle;
+  private double startDistance;
+  private final SlewRateLimiter speedSlewRateLimiter =
+      new SlewRateLimiter(DriveConstants.SLEW_RATE_LIMITER / 2);
+  private double maxSpeed = 0; // in meters/sec
+
+  // Change this to match the name of your camera
+  PhotonCamera camera = new PhotonCamera("photonvision");
+
+  /**
+   * The constructor for the Drive Straight PID command.
+   *
+   * @param drivetrain The drive train subsystem.
+   * @param distance The distance (in meters) the robot needs to cover.
+   */
+  public DriveVision(DriveTrain drivetrain, double distance) {
+    this.drivetrain = drivetrain;
+    this.distance = distance;
+    distancePid =
+        new PIDController(
+            Robot.getDriveTrainConstant("DISTANCE_P").asDouble(),
+            Robot.getDriveTrainConstant("DISTANCE_I").asDouble(),
+            Robot.getDriveTrainConstant("DISTANCE_D").asDouble());
+    straightPid =
+        new PIDController(
+            Robot.getDriveTrainConstant("STRAIGHT_P").asDouble(),
+            Robot.getDriveTrainConstant("STRAIGHT_I").asDouble(),
+            Robot.getDriveTrainConstant("STRAIGHT_D").asDouble());
+    distancePid.setTolerance(Robot.getDriveTrainConstant("DISTANCE_PID_TOLERANCE").asDouble());
+    straightPid.enableContinuousInput(0, 360);
+    addRequirements(drivetrain);
+    // Use addRequirements() here to declare subsystem dependencies.
+  }
+
+  /**
+   * The constructor for the Drive Straight PID command.
+   *
+   * @param drivetrain The drive train subsystem.
+   * @param distance The distance (in meters) the robot needs to cover.
+   * @param maxSpeed The max speed the robot should travel
+   */
+  public DriveVision(DriveTrain drivetrain, double distance, double maxSpeed) {
+    this(drivetrain, distance);
+
+    this.maxSpeed = maxSpeed;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    CommandDebug.trace();
+    startAngle = drivetrain.getYaw();
+    startDistance = drivetrain.getAverageDistance();
+    CommandDebug.trace("startAngle: " + startAngle + " distance: " + distance);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    // TODO: use methods on camera object to check if any targets are detected.
+    // Use something like:
+    //    SmartDashboard.putString("hasTargets", "put details about targets here");
+    // to help output information about whether or not targets are found.
+    //
+    // Read these docs/examples to help figure out what to do:
```

Comment:
Use this guide. Its exactly what youre trying to do: https://docs.photonvision.org/en/latest/docs/examples/aimandrange.html